libfreerdp/core/window.c

```diff
@@ -329,9 +335,14 @@ BOOL update_read_window_state_order(wStream* s, WINDOW_ORDER_INFO* orderInfo, WI
 		Stream_Read_UINT16(s, windowState->numVisibilityRects); /* numVisibilityRects (2 bytes) */
 
 		size = sizeof(RECTANGLE_16) * windowState->numVisibilityRects;
-		windowState->visibilityRects = (RECTANGLE_16*) malloc(size);
-		if (!windowState->visibilityRects)
+		newRect = (RECTANGLE_16*)realloc(windowState->visibilityRects, size);
+		if (!newRect)
```

Comment:
@weizhenwei no problem ;) - We are happy about the improvements. Just wanted to mention it that we have it in one place.